src/monitoring/dp_event.c

```diff
@@ -88,6 +85,12 @@ void dp_process_event_link_msg(struct rte_mbuf *m)
 	}
 
 	port->link_status = status;
+	DPS_LOG_INFO("PF link state changed", DP_LOG_LINKSTATE(port->link_status), DP_LOG_PORT(port));
+
+	if (status == RTE_ETH_LINK_UP)
```

Comment:
I think we should move this part to `dp_link_status_change_event_callback`. For me this looks like an unnecessary round trip from worker thread back to main thread by calling these functions here and the worker thread directly calls the netlink call which is not so nice as the worker thread should minimize the work which doesnt include packet processing.
Another problem with this approach is that worker thread now will enqueue to the monitoring queue which is actually supposed to be enqueued only by the main thread.
This approach will worsen the problem we already have in the code. monitoring_queue is being enqueued by main thread and by the interrupt thread at the moment which is already a problem. Now we bring a third thread (worker thread) which will enqueue to the monitoring queue. 

So in a nutshell:
- During init of the PF ports we can arm the single call timer run on main thread which will call netlink, re-arm if netlink fails, send mac_neigh event with mac address if netlink doesnt fail. This is already done this way. 
- And in `dp_link_status_change_event_callback` we can arm the single call timer run on main thread if the link state is reported as up and let the timer callback call netlink, re-arm if netlink fails, send mac_neigh event with  mac address if netlink doesnt fail. 
- And also in `dp_link_status_change_event_callback` stop the the timer, if the link state gets reported as "down".

Reply:
I moved the indicated code away from worker thread to the "interrupt" thread (the one where timer fires).

I think this should be enough, but if we only ever want to call the MAC retrieval code from one thread (the interrupt thread) and not the main thread, it would require removing the ability to "get the MAC immediately" in the init phase, thus always waiting at least a second for it. Which I do not believe is needed as the real problem here was the worker doing unnecessary stuff, which is fixed now.

Reply:
I also think that the change is sufficient for this PR.